File: frontend/lib/models/expense_model.dart
class ExpenseCreate {
  final double amount;
  final String? category;
  final String? vendorName;
  final String? description;
  final DateTime expenseDate;
  final bool isPaid;

  ExpenseCreate({
    required this.amount,
    this.category,
    this.vendorName,
    this.description,
    required this.expenseDate,
    required this.isPaid,
  });

  Map<String, dynamic> toJson() {
    return {
      'amount': amount,
      'category': category,
      'vendor_name': vendorName,
      'description': description,
      'expense_date': expenseDate.toIso8601String().split('T')[0],
      'is_paid': isPaid,
    };
  }
}
class Expense {
  final int id;
  final int businessId;
  final double amount;
  final String? category;
  final String? vendorName;
  final String? description;
  final DateTime expenseDate;
  final bool isPaid;

  Expense({
    required this.id,
    required this.businessId,
    required this.amount,
    this.category,
    this.vendorName,
    this.description,
    required this.expenseDate,
    required this.isPaid,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['id'],
      businessId: json['business_id'],
      amount: (json['amount'] as num).toDouble(),
      category: json['category'],
      vendorName: json['vendor_name'],
      description: json['description'],
      expenseDate: DateTime.parse(json['expense_date']),
      isPaid: json['is_paid'] ?? false,
    );
  }
}

File: backend/app/models/expense_model.py
from sqlalchemy import Column, Integer, Float, String, DateTime, ForeignKey, Date, Boolean
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from database import Base

class Expense(Base):
    __tablename__ = "expenses"

    id = Column(Integer, primary_key=True)
    business_id = Column(Integer, ForeignKey("businesses.id"), nullable=False, index=True)
    category = Column(String)
    vendor_name = Column(String, nullable=True)
    amount = Column(Float)
    description = Column(String, nullable=True)
    expense_date = Column(Date, nullable=False)
    is_paid = Column(Boolean, default=True)
    created_at = Column(DateTime, server_default=func.now())

    business = relationship("Business", back_populates="expenses")
    payments = relationship("Payment", back_populates="expenses")

